File: server/router/cloudpods/image/image.go
package image

import (
	v1 "ygpt/server/api/v1"

	"github.com/gin-gonic/gin"
)

type ImageRouter struct {
}

func (s *ImageRouter) InitStoImageRouter(Router *gin.RouterGroup) {
	dataRouter := Router.Group("cloud/image/image")
	imageApi := v1.ApiGroupApp.CloudApiGroup.ImageApiGroup.ImageApi
	{

		dataRouter.POST("getImageList", imageApi.GetImageList)           // 获取镜像列表
		dataRouter.POST("getImageByName", imageApi.GetImageByName)       // 获取单个镜像
		dataRouter.DELETE("deleteImage", imageApi.DeleteImage)           // 删除镜像
		dataRouter.DELETE("deleteImageByIds", imageApi.DeleteImageByIds) // 删除镜像
		dataRouter.POST("uploadImage", imageApi.UploadImage)             // 镜像上传
		dataRouter.PUT("updateImageInfo", imageApi.UpdateImageInfo)      // 修改镜像信息
		dataRouter.GET("downloadImage", imageApi.DownloadImage)          // 镜像下载
		dataRouter.POST("betterUpImage", imageApi.BetterUpImage)
		dataRouter.POST("getRecycleImageList", imageApi.GetRecycleImageList) //获取回收站列表
		dataRouter.POST("clearRecycleImage", imageApi.ClearRecycleImage)     // 清除回收镜像
		dataRouter.POST("recoverRecycleImage", imageApi.RecoverRecycleImage) // 恢复回收镜像
		dataRouter.POST("updateImageTag", imageApi.UpdateImageTag)           // 镜像标签修改
	}
}

File: web/src/api/cloudpods/image/image.js
import request from '@/utils/request'

// 镜像标签修改
export function updateImageTag(params) {
    return request({
        url: 'cloud/image/image/updateImageTag',
        method: 'post',
        data: params
    })
}
// 恢复回收镜像
export function recoverRecycleImage(params) {
    return request({
        url: 'cloud/image/image/recoverRecycleImage',
        method: 'post',
        data: params
    })
}
// 清除回收镜像
export function clearRecycleImage(params) {
    return request({
        url: 'cloud/image/image/clearRecycleImage',
        method: 'post',
        data: params
    })
}
// 获取回收站列表
export function getRecycleImageList(params) {
    return request({
        url: 'cloud/image/image/getRecycleImageList',
        method: 'post',
        data: params
    })
}
// 获取镜像列表
export function getImageList(params) {
    return request({
        url: 'cloud/image/image/getImageList',
        method: 'post',
        data: params
    })
}
// 获取单个镜像
export function getImageByName(params) {
    return request({
        url: 'cloud/image/image/getImageByName',
        method: 'post',
        data: params
    })
}
// 上传镜像
export function uploadImage(params, config = {}) {
    return request({
        url: 'cloud/image/image/uploadImage',
        method: 'post',
        data: params,
        // 添加支持进度监听的配置
        onUploadProgress: config.onUploadProgress,
        cancelToken: config.cancelToken
    })
}
// 上传镜像
export function deleteImage(params) {
    return request({
        url: 'cloud/image/image/deleteImage',
        method: 'delete',
        data: params
    })
}
// 上传镜像
export function deleteImageByIds(params) {
    return request({
        url: 'cloud/image/image/deleteImageByIds',
        method: 'delete',
        data: params
    })
}

export function getImagesByRepository(data) {
    return request({
        url: 'cloud/image/imageRepos/getImageReposImageList',
        method:'post',
        data: data 
    })
}